mcweb/backend/sources/tasks.py

```diff
@@ -315,3 +319,113 @@ def schedule_scrape_source(source_id, user):
+                    continue
+
+                earliest_month = min(counts_by_month)
+                first_story = dt.datetime.combine(earliest_month, dt.datetime.min.time())
+                if first_story:
+                    first_story = timezone.make_aware(first_story)
+                    source.first_story = first_story
+                    logger.info("Analyzed source %s. First story publication date: %s" % (source.name, first_story))
+
+            source.modified_at = timezone.now()
+            updated_sources.append(source)
+        except Exception as e:
+            logger.error("Failed to analyze source %s: %s" % (source.name, str(e)))
+
+    if updated_sources:
+        field_name = 'primary_language' if task_name == "update_source_language" else 'first_story'
+        Source.objects.bulk_update(updated_sources, [field_name, "modified_at"])
+        logger.info("Completed %s extraction. Updated %d sources." % (task_name, len(updated_sources)))
+
+    return updated_sources
+
+
+@background(queue=SYSTEM_SLOW)
+def update_source_language(provider_name:str, batch_size: int = 100 ) -> None:
+    six_months_ago = timezone.now() - dt.timedelta(days=SOURCE_UPDATE_DAYS_BACK)
+    while True:
+        sources_for_language = Source.objects.filter(
+            Q(primary_language__isnull=True),
+            name__isnull=False,
+            modified_at__lt=six_months_ago
+        ).order_by("modified_at")[:batch_size]
+
+        if not sources_for_language:
+            logger.info("No new sources to process for language analysis.")
+            break
+
+        updated_sources = analyze_sources(provider_name, sources_for_language, batch_size, six_months_ago, "update_source_language")
+        if updated_sources:
+            logger.info("Successfully updated %d sources for language analysis.", len(updated_sources))
+        else:
+            logger.info("No sources were updated during language analysis.")
+
+@background(queue=SYSTEM_SLOW)
+def update_publication_date(provider_name:str, batch_size: int = 100) -> None:
+    six_months_ago = timezone.now() - dt.timedelta(days=SOURCE_UPDATE_DAYS_BACK)
+    while True:
+        sources_for_publication_date = Source.objects.filter(
+            Q(first_story__isnull=True),
+            name__isnull=False,
+            modified_at__lt=six_months_ago
+        ).order_by("modified_at")[:batch_size]
+
+        if not sources_for_publication_date:
+            logger.info("No new sources to process for publication date analysis.")
+            break
+
+        updated_sources = analyze_sources(provider_name, sources_for_publication_date, batch_size, SOURCE_UPDATE_START_DATE, "update_publication_date")
+        if updated_sources:
+            logger.info("Successfully updated first story for %d sources.", len(updated_sources))
+        else:
+            logger.info("No sources were updated for first story publication date.")
```

Comment:
`analyze_sources` already checks `if updated_sources` and logs accordingly; not sure if we need another.